ts/features/lollipop/hooks/useLollipopLoginSource.tsx

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useState } from "react";
+import { useIOSelector } from "../../../store/hooks";
+import { LoggedOutWithIdp } from "../../../store/reducers/authentication";
+import { isLollipopEnabledSelector } from "../../../store/reducers/backendStatus";
+import { taskGetPublicKey } from "../../../utils/crypto";
+import { getIdpLoginUri } from "../../../utils/login";
+import { lollipopSelector } from "../store/reducers/lollipop";
+import { LoginSourceAsync } from "../types/LollipopLoginSource";
+
+type Props = {
+  loggedOutWithIdpAuth?: LoggedOutWithIdp;
+};
+
+const DEFAULT_LOLLIPOP_HASH_ALGORITHM = "sha256";
+
+export const useLollipopLoginSource = ({ loggedOutWithIdpAuth }: Props) => {
+  const [loginSource, setLoginSource] = useState<LoginSourceAsync>({
+    kind: "initial"
+  });
+
+  const isLollipopEnabled = useIOSelector(isLollipopEnabledSelector);
+  const lollipopState = useIOSelector(lollipopSelector);
```

Comment:
We can use the lollipopKeyTagSelector directly here.

Shall we also change it in order to leverage the capabilities of the Option type of fp-ts? Instead of returning a PublicKey | undefined, let's return an Option<PublicKey>